### PR TITLE
:sparkles: Adding aws cluster-arn validation to klusterlet CRD

### PR DESCRIPTION
**Pull request by jaswalkiranavtar**
## Related issue(s)

Fixes # https://github.com/open-cluster-management-io/ocm/issues/514

### CODE REVIEW

File: test/integration/api/klusterlet_test.go [hunk 1]
```diff
@@ -37,6 +37,23 @@ var _ = Describe("Create Klusterlet API", func() {
...
+				RegistrationDriver: operatorv1.RegistrationDriver{
+					AuthType: "awsirsa",
+					AwsIrsa: &operatorv1.AwsIrsa{
+						ManagedClusterArn: "arn:aws:bks:us-west-2:123456789012:cluster/managed-cluster1",
+						HubClusterArn:     "arn:aws:eks:us-west-2:123456789012:cluster/hub-cluster1",
+					},
+				},
+			}
+			_, err := operatorClient.OperatorV1().Klusterlets().Create(context.TODO(), klusterlet, metav1.CreateOptions{})
+			fmt.Println(err)
```
Comment (qiujian16) [inline test/integration/api/klusterlet_test.go]: this line is not needed.

File: test/integration/api/klusterlet_test.go [hunk 2]
```diff
@@ -37,6 +37,23 @@ var _ = Describe("Create Klusterlet API", func() {
...
+			Expect(err).NotTo(BeNil())
+		})
+	})
```
Comment (qiujian16) [inline test/integration/api/klusterlet_test.go:55]: thanks for adding the test. Would you also add a valid arn to ensure the correct format can pass the validation.

File: test/integration/api/klusterlet_test.go (hunk 1 above)
Comment (jaswalkiranavtar) [reply test/integration/api/klusterlet_test.go]: removed

File: test/integration/api/klusterlet_test.go (hunk 2 above)
Comment (jaswalkiranavtar) [reply test/integration/api/klusterlet_test.go:55]: added

**Pull request merged**